client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx

```diff
@@ -74,6 +75,13 @@ const getTitanMenuItems = ( { mailbox, showRemoveMailboxDialog, translate } ) =>
 			title: translate( 'View Mail', {
 				comment: 'View the Email application (i.e. the webmail) for Titan',
 			} ),
+			onClick: () => {
```

Comment:
It feels like there is a lot of repeated boilerplate in this list of menu items. It might be worth using a local function to generate the click handlers, e.g.
```
const getTitanClickHandler = ( app ) => {
	return () => {
		recordEmailActionTrackEvent( {
			app,
			context: 'email-management-menu',
			provider: 'titan',
		} );
	};
};
...
// use the function to generate the onClick handler
onClick: getTitanClickHandler( 'webmail' ),
```

You could even create a general function that accepts `app` and `provider` if you really want to centralise, and have the `getTitanClickHandler()` function call that function with `( app, 'titan' )`. (But that might be overkill! 😁 )

Reply:
Cool, I agree. It could be refactored. I've created two functions one for Google and another for Titan